hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/RecordListBuilder.java

```diff
@@ -376,7 +383,31 @@ public void revertChildrenOf(@NonNull final SingleTransactionRecordBuilderImpl r
                 followingChildRemoved = true;
             } else {
                 if (child.reversingBehavior() == ReversingBehavior.REVERSIBLE && SUCCESSES.contains(child.status())) {
+                    // clear "side effect" fields
+                    child.accountID(AccountID.DEFAULT);
+                    child.contractID(ContractID.DEFAULT);
+                    child.fileID(FileID.DEFAULT);
+                    child.tokenID(TokenID.DEFAULT);
+                    child.scheduleID(ScheduleID.DEFAULT);
+                    child.scheduledTransactionID(TransactionID.DEFAULT);
+                    child.serialNumbers().clear();
+                    child.topicRunningHash(Bytes.EMPTY);
+                    child.newTotalSupply(-1L);
```

Comment:
This should be `0`, right? Otherwise we would externalize a new total supply of `-1`, which doesn't seem correct.